### PR TITLE
Fix long titles in collections push out the last-edited-by and last-edited-at columns

### LINKED ISSUE #16991: Long titles in collections push out the last-edited-by and last-edited-at columns
**To Reproduce**
1. Create a question or a dashboard with a long name. Example `this question has a very very very very very very very very very very very looooooooooooooooooooong name`
2. Go to `/collection/root` (or a collection a new item is saved to)
3. Notice the table doesn't fit and "last edited" column are pushed far to the right

**Expected behavior**
The table has to have a fixed width and the title has to be truncated

**Screenshots**

CleanShot 2021-07-12 at 21 39 42@2x

**Information about your Metabase Installation:**
Metabase 0.40

### PR DESCRIPTION
**Pull request by kulyk**
Fixes #16991

### To Verify

1. Create a question or a dashboard with a long name. Example `this question has a very very very very very very very very very very very looooooooooooooooooooong name`
2. Go to `/collection/root` (or a collection a new item is saved to)
3. Make sure the table has a fixed width and the long title is truncated

### Demo

**Before**

CleanShot 2021-07-12 at 21 39 42@2x

**After**

https://user-images.githubusercontent.com/17258145/125340169-d11f8e00-e35a-11eb-906e-5adf08130ca7.mov

### CODE REVIEW

File: frontend/src/metabase/collections/components/BaseItemsTable.jsx [hunk 1]
```diff
@@ -112,13 +113,13 @@ function BaseItemsTable({
     });
 
   return (
-    <table className="ContentTable" {...props}>
+    <Table {...props}>
       <colgroup>
-        <col span="1" style={{ width: "5%" }} />
-        <col span="1" style={{ width: "64%" }} />
+        <col span="1" style={{ width: "60px" }} />
```
Comment (gusaiani) [inline frontend/src/metabase/collections/components/BaseItemsTable.jsx:118]: Remove `span` attributes if they are all set to `1`? 🤔

**Review by gusaiani — APPROVED** [review]
Nice html table chops 👍

One inlined comment.

File: frontend/src/metabase/collections/components/BaseItemsTable.jsx (hunk 1 above)
Comment (kulyk) [reply frontend/src/metabase/collections/components/BaseItemsTable.jsx]: Good call, fixed. Thank you!

**Comment by mazameli** [comment]
Overall this seems great. 👍🏼 

One non-blocking suggestion for improvement: I think the last-edited-by and last-edited-at columns could have _fixed_ horizontal space between them, and can hug the right side of the screen. Right now the space increases proportionately as the screen width increases, and I think that's excessive and can unnecessarily use up space that could be used for an item's title.

Here's the current behavior:

column spacing

By contrast, Google Drive's columns behave more like what I'm suggesting:

drive's behavior

**Comment by kulyk** [comment]
Thanks for your review @mazameli 

> One non-blocking suggestion for improvement: I think the last-edited-by and last-edited-at columns could have _fixed_ horizontal space between them, and can hug the right side of the screen. Right now the space increases proportionately as the screen width increases, and I think that's excessive and can unnecessarily use up space that could be used for an item's title.

If we fix width for last-edited-by and last-edited-at, they will overlap the "Name" column at some point. See an example:

https://user-images.githubusercontent.com/17258145/125488935-0620853e-56d3-46cf-98b1-97ec5addaafe.mov

Google Drive hides some columns for smaller screens one-by-one. What if we do the same thing? We'll need to hide "last edited at" and "last edited by" anyway for mobile layouts, so maybe we can hide them a bit earlier? See the demo:

https://user-images.githubusercontent.com/17258145/125489277-c0a7549a-cf2a-4811-b938-7e6e6c32fbfc.mov

**Comment by mazameli** [comment]
> Google Drive hides some columns for smaller screens one-by-one. What if we do the same thing? We'll need to hide "last edited at" and "last edited by" anyway for mobile layouts, so maybe we can hide them a bit earlier? See the demo:

This makes sense to me and your demo looks good. 👍🏼

**Comment by kulyk** [comment]
Thank @mazameli! Pushing the demo code here then

**Pull request merged**